libraries/WiFi/src/WiFiSTA.cpp

```diff
@@ -215,7 +215,7 @@ bool WiFiSTAClass::bandwidth(wifi_bandwidth_t bandwidth) {
  * @return true if STA is connected to an AP
  */
 bool WiFiSTAClass::isConnected() {
-  return STA.connected();
+  return STA.connected() && STA.hasIP();
```

Comment:
just `STA.hasIP()` is enough here. You can not have IP if you are not connected.

Reply:
Even if it is comfigured as static IP?

Reply:
you are correct. `hasIP()` checks if either IP has been assigned or that it has static. Since even when static `GOT_IP` event is received, maybe worth changing there too. Not sure what is more correct for `gotIP()`, but we can discuss that later.